### PR TITLE
Docs enhancement

### PR DESCRIPTION
**Pull request by SOHELAHMED7**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | -

Docs enhancement

### CODE REVIEW

File: docs/guide/tutorial-yii-integration.md [hunk 1]
```diff
@@ -114,6 +118,24 @@ Like in a Yii application, you should configure the application instance based o
...
+'components' => [
+    'radar' => require __DIR__.'/../../app/My/MyClass.php'
+    
+    
+// MyClass.php
+namespace App\Custom;
+
+use \some\class\Here;
+
+return new MyClass  extends \yii\base\BaseObject // <-- annonymous class is supported in PHP >= 7
```
Comment (samdark) [inline docs/guide/tutorial-yii-integration.md]: That looks interesting. What's the purpose of doing that instead of using regular class names?

**Comment by samdark** [comment]
Sorry, @SOHELAHMED7 but I don't get the idea. What do you want to achieve with the code in these changes?

**Comment by SOHELAHMED7** [comment]
Yii version - 2.0.32-dev (2.0.* can be considered)
Laravel version - 5.5.1 (5.*  can be considered)
PHP 7.2

Use case:
I want to use a new created class (e.g. Radar.php) that extends a Yii class (e.g. \yii\base\BaseObject) in a Laravel app

In Laravel app, typically  code stays in laravel-project/app (https://github.com/laravel/quickstart-basic/tree/master/app) folder

In our case
file path - laravel-project/app/Custom/Radar.php
namespace - App\Custom\Radar.php

and it's composer autoload setting:
https://github.com/laravel/quickstart-basic/blob/master/composer.json#L23

path of yii web config file
laravel-project/config/yii/web.php


Now in both below case with Radar.php have class as 
`class Radar extends \yii\base\BaseObject`
lambda

```php
'radar' => # require __DIR__.'/../../app/Custom/Radar.php'
            function () {
                return new \App\Custom\Radar(['cookieValidationKey' => '5o3rYSgu3Nxsdfsdf67W3424uOy6yUtAuQpjCtwQFn']);
            }
```




array
```php
'radar' => # require __DIR__.'/../../app/Custom/Radar.php'
            // function () {
            //     return new \App\Custom\Radar(['cookieValidationKey' => '5o3rYSgu3Nxsdfsdf67W3424uOy6yUtAuQpjCtwQFn']);
            // }
            [
                'class' => \App\Custom\Radar::class,
                'cookieValidationKey' => 'as6d4as64d65asd64a65sd'
            ]
```

I got error - 
 yii\base\InvalidConfigException: Failed to instantiate component or class "App\Custom\Radar". in laravel-project/vendor/yiisoft/yii2/di/Container.php:449


when I did 

```php
'components' => [
        'radar' => require __DIR__.'/../../app/Custom/Radar.php'
```
with Radar.php as 

`return new class extends \yii\base\BaseObject` 

I got no error and it went all successfull

Glad to elaborate more if needed

File: docs/guide/tutorial-yii-integration.md (hunk 1 above)
Comment (SOHELAHMED7) [reply docs/guide/tutorial-yii-integration.md]: https://github.com/yiisoft/yii2/pull/17795#issuecomment-573321975

**Comment by rob006** [comment]
It looks like namespace issue (`App\Custom\Radar.php` is definitely incorrect namespace), so `App\Custom\Radar` cannot be autoloaded.

**Comment by samdark** [comment]
The issue could be solved better with composer autoloader or with registering Yii autoloader. The way provided, while works, isn't a good practice. Sorry but won't merge the pull request. And thanks for contributing anyway.

**Comment by SOHELAHMED7** [comment]
Oh. I later realized that I cannot use that annonymous class in Laravel by `use` namespace. Sorry.

>The issue could be solved better with composer autoloader or with registering Yii autoloader.

Can you provide  any link to any online relavant content

**Comment by samdark** [comment]
https://www.yiiframework.com/doc/guide/2.0/en/tutorial-yii-integration#using-yii-in-others

**Pull request closed without merging**